src/ufuncs.c

```diff
@@ -1363,7 +1363,7 @@ static void create_collection_func(char **args, npy_intp *dimensions,
 {
     GEOSGeometry *g, *g_copy;
     int n_geoms, type;
-    char actual_type, expected_type;
+    char actual_type, first_expected_type, last_expected_type;
```

Comment:
@caspervdw do you foresee cases where testing against a broader range of geometry types will be useful?  As written, testing this as a range between first and last expected type feels awkward (it just so happens that these are in sequence), when there is at most 2 possible expected types.  I think equality would be better in this case.


Since you compare against them below, I'd suggest initializing all to `-1`.

```suggestion
    char actual_type = -1;
    char expected_type = -1;
    char alt_expected_type = -1;
```

Then, I think you could drop the assignments to the same types on lines 1378, 1386 below, and replace 1381-1382 with
```
expected_type = GEOS_LINESTRING;
alt_expected_type = GEOS_LINEARRING;
```

To make the final comparison on 1403 a bit more straightforward:
```
if ((actual_type != expected_type) & (actual_type!= alt_expected_type)) {
```

Reply:
That's better indeed, thanks for the suggestion.

The `alt_expected_type` has to be set in each iteration though.